### PR TITLE
Feature/query messages with attachments

### PR DESCRIPTION
**Pull request by jkoenig134**
## Summary by CodeRabbit

- **New Features**
	- Enhanced message querying to handle specific input values related to attachments.
- **Tests**
	- Updated test utilities to support message exchange with attachments.
	- Added new tests for querying messages with attachments.

### CODE REVIEW

File: packages/runtime/test/lib/testUtils.ts [hunk 1]
```diff
@@ -235,15 +235,8 @@ export async function exchangeMessage(transportServicesCreator: TransportService
...
-    const recipientAddress = (await getRelationship(transportServicesCreator)).peer;
-    const messageId = (await sendMessage(transportServicesCreator, recipientAddress, undefined, [file.id])).id;
-    const messages = await syncUntilHasMessages(transportServicesRecipient);
-    expect(messages).toHaveLength(1);
-
-    const message = messages[0];
-    expect(message.id).toStrictEqual(messageId);
 
-    return message;
+    return await exchangeMessage(transportServicesCreator, transportServicesRecipient, [file.id]);
```
[the remark this answers is not in the document]
Comment (jkoenig134) [reply packages/runtime/test/lib/testUtils.ts:239]: Our lint rules and code standards require this await

File: packages/runtime/test/transport/messages.test.ts [hunk 2]
```diff
@@ -189,4 +190,13 @@ describe("Message query", () => {
...
     });
+
+    test("query Messages withAttachments", async () => {
+        await exchangeMessageWithAttachment(transportServices1, transportServices2);
+        await exchangeMessage(transportServices1, transportServices2);
+
+        const messages = await transportServices2.messages.getMessages({ query: { attachments: "+" } });
+
+        expect(messages.value.every((m) => m.attachments.length > 0)).toBe(true);
+    });
```
Comment (tnotheis) [inline packages/runtime/test/transport/messages.test.ts:201]: That test is weird. :) I would save the two messages and in the assertion, I would do the following assertions:

```c#
expect(messages).toHaveLength(1);
expect(messages[0].id).toBe(messageWithAttachments.Id)
```

**Review by tnotheis — CHANGES_REQUESTED** [review]

File: packages/runtime/test/transport/messages.test.ts (hunk 2 above)
Comment (jkoenig134) [reply packages/runtime/test/transport/messages.test.ts:201]: actually I can't rely on having an empty list of messages when there were other tests before.

Comment (jkoenig134) [reply packages/runtime/test/transport/messages.test.ts:201]: But I can add the second check, that the first message is contained in the array.

Comment (tnotheis) [reply packages/runtime/test/transport/messages.test.ts:201]: Ahh, I expected the transportServices to be reinitialized before each test.

Comment (jkoenig134) [reply packages/runtime/test/transport/messages.test.ts]: Is it better now?

**Review by sebbi08 — APPROVED** [review]

**Review by tnotheis — APPROVED** [review]

**Pull request merged**